### ISSUE TITLE
Segementation Fault When Reading File With Java App

### ISSUE THREAD

**Issue by jeloba** [metalbear-co/mirrord#2001]
### Bug Description

Reading a file in a Java app (running in a VS Code debugger) triggers a segmentation fault:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f33526e66aa, pid=127287, tid=127295
#
# JRE version: OpenJDK Runtime Environment (17.0.8.1+1) (build 17.0.8.1+1-Ubuntu-0ubuntu122.04)
# Java VM: OpenJDK 64-Bit Server VM (17.0.8.1+1-Ubuntu-0ubuntu122.04, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
# Problematic frame:
# C  [libc.so.6+0xe66aa]  readdir64+0x2a
#
# No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# hs_err_pid127287.log
#
# If you would like to submit a bug report, please visit:
#   Unknown
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```
The config in use contains the following stanza:

```
{
    "feature": {
        "fs": {
          "mode": "read",
          "read_only": [ "/etc/*" ]
        }
    }
}
```

### Steps to Reproduce

1. configure and enable mirrord
2. start a debugging session

### Backtrace

```shell
Current thread (0x00007f3344046e10):  JavaThread "main" [_thread_in_native, id=127295, stack(0x00007f3350f00000,0x00007f3351000000)]

Stack: [0x00007f3350f00000,0x00007f3351000000],  sp=0x00007f3350ffdfd0,  free space=1015k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [libc.so.6+0xe66aa]  readdir64+0x2a

Java frames: (J=compiled Java code, j=interpreted, Vv=VM code)
j  sun.nio.fs.UnixNativeDispatcher.readdir(J)[B+0 java.base@17.0.8.1
j  sun.nio.fs.UnixDirectoryStream$UnixDirectoryIterator.readNextEntry()Ljava/nio/file/Path;+52 java.base@17.0.8.1
j  sun.nio.fs.UnixDirectoryStream$UnixDirectoryIterator.hasNext()Z+16 java.base@17.0.8.1
j  java.nio.file.FileTreeWalker.next()Ljava/nio/file/FileTreeWalker$Event;+37 java.base@17.0.8.1
j  java.nio.file.Files.walkFileTree(Ljava/nio/file/Path;Ljava/util/Set;ILjava/nio/file/FileVisitor;)Ljava/nio/file/Path;+255 java.base@17.0.8.1
j  java.nio.file.Files.walkFileTree(Ljava/nio/file/Path;Ljava/nio/file/FileVisitor;)Ljava/nio/file/Path;+11 java.base@17.0.8.1
[...more]
```

### Your operating system and version

Ubuntu 22.04.3 LTS

### Local process

Java debugger in VS Code

### Additional Info

For additional context: 

* The location that is accessed is provided by the Vault agent. So it's not located on the main file system. 
* Listing all files in "/" returns results form the local system.
* Listing all files in "/etc" causes a crash.

**Comment by Razz4780** [comment]
We found the source of the bug and fixed it in #2003 
All should work with the next release :)

**Comment by jeloba** [comment]
I gave it a try and the issue is resolved. Thanks @Razz4780 and @eyalb181 !
